src/coreComponents/mesh/generators/VTKUtilities.cpp

```diff
@@ -190,72 +196,143 @@ vtkSmartPointer< vtkMultiProcessController > getController()
   return controller;
 }
 
+VTKMeshExtension stringToVTKMeshExtension( const string & extension )
+{
+  if( extension == "vtk" )
+  {
+    return VTKMeshExtension::vtk;
+  }
+  else if( extension == "vtu" )
+  {
+    return VTKMeshExtension::vtu;
+  }
+  else if( extension == "vtr" )
+  {
+    return VTKMeshExtension::vtr;
+  }
+  else if( extension == "vts" )
+  {
+    return VTKMeshExtension::vts;
+  }
+  else if( extension == "vti" )
+  {
+    return VTKMeshExtension::vti;
+  }
+  else if( extension == "pvtu" )
+  {
+    return VTKMeshExtension::pvtu;
+  }
+  else if( extension == "pvtr" )
+  {
+    return VTKMeshExtension::pvtr;
+  }
+  else if( extension == "pvts" )
+  {
+    return VTKMeshExtension::pvts;
+  }
+  else if( extension == "pvti" )
+  {
+    return VTKMeshExtension::pvti;
+  }
+  else
+  {
+    return VTKMeshExtension::invalid;
+  }
+}
+
 vtkSmartPointer< vtkDataSet >
 loadMesh( Path const & filePath )
 {
   string const extension = filePath.extension();
   vtkSmartPointer< vtkDataSet > loadedMesh;
 
-  if( extension == "pvtu" )
+  auto const parallelRead = [&]( auto const vtkGridReader )
   {
-    auto const vtkUgReader = vtkSmartPointer< vtkXMLPUnstructuredGridReader >::New();
-    vtkUgReader->SetFileName( filePath.c_str() );
-    vtkUgReader->UpdateInformation();
-    vtkUgReader->UpdatePiece( MpiWrapper::commRank(), MpiWrapper::commSize(), 0 );
-    loadedMesh = vtkUgReader->GetOutput();
+    vtkGridReader->SetFileName( filePath.c_str() );
+    vtkGridReader->UpdateInformation();
+    vtkGridReader->UpdatePiece( MpiWrapper::commRank(), MpiWrapper::commSize(), 0 );
+    return vtkGridReader->GetOutput();
+  };
 
-    // TODO: Apply vtkStaticCleanUnstructuredGrid, once it is included in the next VTK release.
-    //       https://gitlab.kitware.com/vtk/vtk/-/blob/master/Filters/Core/vtkStaticCleanUnstructuredGrid.h
-    //       This removes duplicate points, either present in the dataset, or resulting from merging pieces.
-  }
-  else if( extension == "pvts" )
+  auto const serialRead = [&]( auto const vtkGridReader )
   {
-    auto const vtkSgReader = vtkSmartPointer< vtkXMLPStructuredGridReader >::New();
-    vtkSgReader->SetFileName( filePath.c_str() );
-    vtkSgReader->UpdateInformation();
-    vtkSgReader->UpdatePiece( MpiWrapper::commRank(), MpiWrapper::commSize(), 0 );
-    loadedMesh = vtkSgReader->GetOutput();
-  }
-  else
+    vtkGridReader->SetFileName( filePath.c_str() );
+    vtkGridReader->Update();
+    return vtkGridReader->GetOutput();
```

Comment:
You could merge the rank check into this function, something like
```suggestion
    if( MpiWrapper::commRank() == 0 )
    {    
         vtkGridReader->SetFileName( filePath.c_str() );
         vtkGridReader->Update();
         return vtkGridReader->GetOutput();
    }
    else
    {
        using GridType = TYPEOFPTR( vtkGridReader->GetOutput() );
        return vtkSmartPointer< GridType >::New();
    }
```

Reply:
Thank you for the suggestion. Done.